lib/Serge/Engine/Plugin/parse_yaml.pm

```diff
@@ -134,12 +141,30 @@ sub parse {
         die $error_text;
     }
 
+    if ($self->{data}->{yaml_mode} eq 'rails') {
+        my @tree_keys = keys(%$tree);
+        my $tree_count = int(@tree_keys);
+        if ($tree_count == 0) {
+            # Special case NOOP. Empty data
+        } elsif ($tree_count == 1) {
+            $tree = $tree->{$tree_keys[0]};
+        } else {
+            my $err = join(', ', @tree_keys);
+            $err = "YAML file with more than 1 key at root in yaml_mode: rails: $err";
+            $self->{errors}->{$self->{parent}->{engine}->{current_file_rel}} = $err;
+            # Or should we die?
+            warn "WARNING: $err";
+        }
+    }
+
     # Process tree recursively
 
     $self->process_node('', $tree, $callbackref, $lang);
 
     # Reconstruct YAML
 
+    $tree = {$lang => $tree} if ($lang && $self->{data}->{yaml_mode} eq 'rails');
```

Comment:
I will confess I’m still not sure I am 100% clear on Serge’s use of the terms language, locale, & culture. 

What Rails expects is what the config files mark as `%CULTURE%`, however by default Rails drops the region/country portion when it is the same as the language. So I think using `Serge::Util::culture_from_lang` and (optionally) dropping the region is probably the way to go.

I don’t know if a separate Util method is warranted here, but let me see what I can do within the parse method.

[Speaking of `culture_from_lang`, I think that the default region/country per language should probably be configurable, since different users will want to map those differently. (Notably `ar` => `ar-AE`, but also `hi` => `hi-IN` is missing). I will file a separate issue for that.]